Description: Fix sockaddr storage for IPv6 addresses.
Forwarded: Fixed in trunk
Author: Stig Sandbeck Mathisen <ssm@debian.org>
Last-Update: 2011-01-21
Index: gearmand-0.14/libgearman-server/gearmand.c
===================================================================
--- gearmand-0.14.orig/libgearman-server/gearmand.c	2010-12-14 10:55:02.444346085 +0100
+++ gearmand-0.14/libgearman-server/gearmand.c	2010-12-14 10:56:00.414345977 +0100
@@ -500,14 +500,14 @@
                           void *arg)
 {
   gearmand_port_st *port= (gearmand_port_st *)arg;
-  struct sockaddr sa;
+  struct sockaddr_storage sa;
   socklen_t sa_len;
   char host[NI_MAXHOST];
   char port_str[NI_MAXSERV];
   int ret;
 
   sa_len= sizeof(sa);
-  fd= accept(fd, &sa, &sa_len);
+  fd= accept(fd, (struct sockaddr *)&sa, &sa_len);
   if (fd == -1)
   {
     if (errno == EINTR)
